Branch-SDK/Branch-SDK/BNCDeviceInfo.m

```diff
@@ -387,4 +410,47 @@ + (NSString*) userAgentString {
     return browserUserAgent();
 }
 
+- (NSDictionary*) v2dictionary {
+    NSMutableDictionary *dictionary = [NSMutableDictionary new];
+
+    #define BNCFieldDefinesDictionaryFromSelf
+    #include "BNCFieldDefines.h"
+
+    addString(osName,               os);
+    addString(osVersion,            os_version);
+    addString(extensionType,        environment);
+    addString(vendorId,             idfv);
+    addString(adId,                 idfa);
+    addString(browserUserAgent,     user_agent);
+    addString(country,              country);
+    addString(language,             language);
+    addString(brandName,            brand);
+    addString(applicationVersion,   app_version);
+    addString(modelName,            model);
+    addDouble(screenScale,          screen_dpi);
+    addNumber(screenHeight,         screen_height);
+    addNumber(screenWidth,          screen_width);
+    addBoolean(unidentifiedDevice,  unidentified_device);
+    addString(localIPAddress,       local_ip);
+    
+    #include "BNCFieldDefines.h"
+
+    if (!self.isAdTrackingEnabled)
+        dictionary[@"limit_ad_tracking"] = CFBridgingRelease(kCFBooleanTrue);
+
+    NSString *s = nil;
+    BNCPreferenceHelper *preferences = [BNCPreferenceHelper preferenceHelper];
+
+    s = preferences.userIdentity;
+    if (s.length) dictionary[@"developer_identity"] = s;
+
+    s = preferences.deviceFingerprintID;
+    if (s.length) dictionary[@"device_fingerprint_id"] = s;
+
+    dictionary[@"sdk"] = @"ios";
```

Comment:
what happens if this is the titanium / cordova / react native / air / whatever SDK?

Reply:
@jdee How do you handle this now?  Do those SDKs just report `ios` now?

I could make a setter for `BNCDeviceInfo` to set the SDK.

Reply:
I don't think the individual SDKs can override that now. It's probably useful to report both that it's iOS and that it's used with RN, Cordova, etc.

Reply:
let's leave as-is for now and address it separately. Thanks!

Reply:
Yeah, I'm pretty sure Cordova only reports iOS or Android at the moment (depending on which underlying SDK it uses).